### PR TITLE
sql: accept interval types in placeholders

### PR DESCRIPTION
**Pull request by madelynnblue**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7382)

### CODE REVIEW

File: sql/pgwire_test.go [hunk 1]
```diff
@@ -1166,3 +1166,43 @@ func TestPGWireOverUnixSocket(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestPGTypes verifies that all types can be sent with prepared statements.
```
Comment (tamird) [inline sql/pgwire_test.go]: what's the difference between this test and `TestPGPreparedExec`?

**Comment by mjibson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[sql/pgwire_test.go, line 1170 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKqWTH3Tea4hbU5F2Qf-r1-1170:-KKt7pHpV2OznqDUhqPz:1214127971) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9c88ae93ded8fc8233837a5ea6655da5f94eef4/sql/pgwire_test.go#L1170)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> what's the difference between this test and `TestPGPreparedExec`?
</details>
Nothing. I thought it'd be more clear to just do this specifically in its own test since it creates a bunch of variables.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by tamird** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[sql/pgwire_test.go, line 1170 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKqWTH3Tea4hbU5F2Qf-r1-1170:-KKtdsMZR2wNyttoGClK:395758317) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9c88ae93ded8fc8233837a5ea6655da5f94eef4/sql/pgwire_test.go#L1170)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary>
> Nothing. I thought it'd be more clear to just do this specifically in its own test since it creates a bunch of variables.
</details>
Seems like it's add just another place to look for placeholder related tests

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by mjibson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/pgwire_test.go, line 1170 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKqWTH3Tea4hbU5F2Qf-r1-1170:-KKvW6l3TMkUohMWqePY:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9c88ae93ded8fc8233837a5ea6655da5f94eef4/sql/pgwire_test.go#L1170)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Seems like it's add just another place to look for placeholder related tests
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by tamird** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[sql/pgwire_test.go, line 768 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKvluwdsX6_XvbLJaxa:-KKvluwdsX6_XvbLJaxb:-1223699974) ([raw file](https://github.com/cockroachdb/cockroach/blob/55608b97ff2b25ee84980f285e9c6db2bd6a3234/sql/pgwire_test.go#L768)):*
> ```Go
> 					"2015-01-02",
> 					time.Now().Format(time.RFC3339Nano),
> 					time.Hour.String(),
> ```

does this not work without stringifying? ditto for date and timestamp

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by mjibson** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 768 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKvluwdsX6_XvbLJaxa:-KKyRUPFr0WF8Fu4r2pO:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/55608b97ff2b25ee84980f285e9c6db2bd6a3234/sql/pgwire_test.go#L768)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> does this not work without stringifying? ditto for date and timestamp
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by mjibson** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 768 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKvluwdsX6_XvbLJaxa:-KKyeOas84Kv5QibxpsK:-1546606769) ([raw file](https://github.com/cockroachdb/cockroach/blob/55608b97ff2b25ee84980f285e9c6db2bd6a3234/sql/pgwire_test.go#L768)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary>
> Done.
</details>
Intervals can't be sent as time.Durations because they send over just an int64, which doesn't parse as a valid interval.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 801 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKyf2VNE35KTsknJLk9:-KKyf2VNE35KTsknJLkA:712607617) ([raw file](https://github.com/cockroachdb/cockroach/blob/5afe0bf93098f83c10f2bfe49a75c0ff37892a57/sql/pgwire_test.go#L801)):*
> ```Go
> 			if result, err := execFunc(test.qargs...); err != nil {
> 				if test.error == "" {
> 					fmt.Println("GOTERR", err, "QUERY", query, "EXPECT", test.error)
> ```

what's up with this?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by mjibson** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 801 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKyf2VNE35KTsknJLk9:-KKyh79_5qIoxuu8bjeh:-2027423687) ([raw file](https://github.com/cockroachdb/cockroach/blob/5afe0bf93098f83c10f2bfe49a75c0ff37892a57/sql/pgwire_test.go#L801)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> what's up with this?
</details>
Debug helper for a bug triggered by this work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by mjibson** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 801 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-KKyf2VNE35KTsknJLk9:-KKyhJTMPt8loRM_eMia:1740281843) ([raw file](https://github.com/cockroachdb/cockroach/blob/5afe0bf93098f83c10f2bfe49a75c0ff37892a57/sql/pgwire_test.go#L801)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary>
> Debug helper for a bug triggered by this work.
</details>
Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7382#-:-KKyhZe_8dWIlSZp9Hzz:1417088721)*

**Pull request merged**